### PR TITLE
StringValidator.Equals checks for null entries

### LINKED ISSUE #473: The StringValidator.Equals should check for null entries in the value list and needs tests
The [StringValidator.Equals.isSpecificationValid()](https://github.com/eeverman/andhow/blob/homepage/andhow-core/src/main/java/org/yarnandtail/andhow/valid/StringValidator.java#L26) method should check for null entries in the array of values.  A null entry in the array should result in `false`.

Additionally, StringValidator.Equals needs unit tests.  All the other [StringValidator.XXX classes have tests](https://github.com/eeverman/andhow/blob/homepage/andhow-core/src/test/java/org/yarnandtail/andhow/valid/StringValidatorTest.java), but not this one.

### PR DESCRIPTION
**Pull request by alex-kar**
Fixes #473 

### All Submissions:

Have you checked that...
* [x] Your pull request is to the *_main_* branch
* [x] Your code is up to date with the latest code from the *_main_*
* [x] You followed the guidelines in our [Developer Guide](https://sites.google.com/view/andhow/developer)?
* [x] Your code does not decrease test coverage (maybe it improves coverage!!)
* [x] If this is related to an issue, please include a link to the issue with the 'Fixes #XXX' syntax.

### CODE REVIEW

File: andhow-core/src/test/java/org/yarnandtail/andhow/valid/StringValidatorTest.java [hunk 1]
```diff
@@ -115,5 +149,85 @@ public void testRegexIsValid() {
...
+	/**
+	 * Test of isSpecificationValid method, of class Equals.
+	 */
+	@Test
+	public void equalsIsSpecificationValidTest() {
+		StringValidator.Equals instance = new StringValidator.Equals((String) null);
+		assertFalse(instance.isSpecificationValid());
+
+		instance = new StringValidator.Equals();
+		assertFalse(instance.isSpecificationValid());
```
Comment (eeverman) [inline andhow-core/src/test/java/org/yarnandtail/andhow/valid/StringValidatorTest.java:180]: Nice catch to test this empty constructor

File: andhow-core/src/test/java/org/yarnandtail/andhow/valid/StringValidatorTest.java [hunk 2]
```diff
@@ -115,5 +149,85 @@ public void testRegexIsValid() {
...
+	}
+
+	/**
+	 * Test of getInvalidSpecificationMessage method, of class Equals.
+	 */
+	@Test
+	public void equalsGetInvalidSpecificationMessageTest() {
+		assertEquals("The Equals list must contain at least one value",
+				new StringValidator.Equals().getInvalidSpecificationMessage());
+	}
```
Comment (eeverman) [inline andhow-core/src/test/java/org/yarnandtail/andhow/valid/StringValidatorTest.java:223]: I think this should be updated to:
`The list must contain at least one value and none of the values can be null`

Referring to 'Equals list' won't make sense to the user any more, now that we have changed the builder method to `oneOf`.

**Review by eeverman — CHANGES_REQUESTED** [review]
Nice work and an improvement of test coverage as well.
One small change to a validation message.

File: andhow-core/src/test/java/org/yarnandtail/andhow/valid/StringValidatorTest.java (hunk 2 above)
Comment (alex-kar) [reply andhow-core/src/test/java/org/yarnandtail/andhow/valid/StringValidatorTest.java]: Updated the message

**Review by eeverman — APPROVED** [review]
Looks great!

**Pull request merged**